scripts/build_test.py

```diff
@@ -647,4 +653,58 @@ def test_generate_delete_tasks_to_remove_deleted_files(self):
         self.assertEqual(
             len(delete_tasks), build.get_file_count(MOCK_TEMPLATES_DEV_DIR))
 
+    def test_compiled_js_dir_validation(self):
+        """Test that build.COMPILED_JS_DIR is validated correctly with
+        outDir in tsconfig.json.
+        """
+        build.test_compiled_js_dir_is_valid()
+
+        out_dir = ''
+        with open('tsconfig.json') as f:
+            config_data = json.load(f)
+            out_dir = os.path.join(config_data['compilerOptions']['outDir'], '')
+        with self.assertRaisesRegexp(
+            Exception,
+            'COMPILED_JS_DIR: %s is does not match the output directory '
+            'in tsconfig.json: %s' % (
+                MOCK_COMPILED_JS_DIR, out_dir)), self.swap(
+                    build, 'COMPILED_JS_DIR', MOCK_COMPILED_JS_DIR):
+            build.test_compiled_js_dir_is_valid()
+
+    def test_compiled_js_dir_is_deleted_before_compilation(self):
+        """Test that compiled_js_dir is deleted before a fresh
+        compilation.
+        """
+        def mock_check_call(unused_cmd):
+            pass
+        # pylint: disable=unused-argument
+        def mock_call(unused_cmd, shell, stdout):
+            pass
+        def mock_popen(unused_cmd, stdout):
+            pass
+        # pylint: enable=unused-argument
+        def mock_test_compiled_js_dir_is_valid():
+            pass
+
+        with self.swap(
+            build, 'COMPILED_JS_DIR', MOCK_COMPILED_JS_DIR), self.swap(
+                build, 'test_compiled_js_dir_is_valid',
+                mock_test_compiled_js_dir_is_valid):
+            if os.path.exists(os.path.dirname(MOCK_COMPILED_JS_DIR)):
```

Comment:
This seems to be affirmatively deleting the compiled_js_dir, which invalidates the point of the test as described in the test name. Shouldn't we instead be ensuring that the dir exists (creating the dir if necessary), and then running the functions, and then verifying that the dir no longer exists?

See points 1 and 2 in [Writing Backend Tests](https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests).

Reply:
Updated.